FindMKL.cmake

```diff
@@ -117,114 +183,130 @@ else()
+                                                                                ${_omp_found_var})
+        endif()
+        
+        if(${_blacs_cpt_requested} GREATER "-1" OR
+           ${_scalapack_cpt_requested} GREATER "-1")
+            find_package_handle_standard_args(MKL_BLACS_${_config} REQUIRED_VARS _mkl_blacs_lib
+                                                                                 MKL_BLAS_${_config}_FOUND
+                                                                                 MPI_FOUND)
+        endif()
+
+        if(${_scalapack_cpt_requested} GREATER "-1")
+            find_package_handle_standard_args(MKL_SCALAPACK_${_config} REQUIRED_VARS _mkl_scalapack_lib
+                                                                                     MKL_BLACS_${_config}_FOUND)
+        endif()
+   
+        if(MKL_BLAS_${_config}_FOUND AND NOT TARGET mkl::blas_${_tgt_config})
+            add_library(mkl::blas_${_tgt_config} INTERFACE IMPORTED)
+            set_target_properties(mkl::blas_${_tgt_config} PROPERTIES
+                INTERFACE_INCLUDE_DIRECTORIES "${MKL_INCLUDE_DIR}"
+                INTERFACE_LINK_LIBRARIES "${_mkl_interface_lib};${_mkl_threading_lib};${MKL_CORE_LIB};${_omp_target};Threads::Threads")
+        endif()
+
+        if(MKL_BLACS_${_config}_FOUND AND NOT TARGET mkl::blacs_${_tgt_config})
+            add_library(mkl::blacs_${_tgt_config} INTERFACE IMPORTED)
+            set_target_properties(mkl::blacs_${_tgt_config} PROPERTIES
+                INTERFACE_INCLUDE_DIRECTORIES "${MKL_INCLUDE_DIR}"
+                INTERFACE_LINK_LIBRARIES "${_mkl_interface_lib};${_mkl_threading_lib};${MKL_CORE_LIB};${_mkl_blacs_lib};${_omp_target};Threads::Threads")
+        endif()
 
-    add_library(mkl::threading UNKNOWN IMPORTED)
-    set_target_properties(mkl::threading PROPERTIES IMPORTED_LOCATION ${MKL_THREADING_LIB})
+        if(MKL_SCALAPACK_${_config}_FOUND AND NOT TARGET mkl::scalapack_${_tgt_config})
+            add_library(mkl::scalapack_${_tgt_config} INTERFACE IMPORTED)
+            set_target_properties(mkl::scalapack_${_tgt_config} PROPERTIES
+                INTERFACE_INCLUDE_DIRECTORIES "${MKL_INCLUDE_DIR}"
+                INTERFACE_LINK_LIBRARIES "${_mkl_scalapack_lib};mkl::blacs_${_tgt_config}")
+        endif()
+    endforeach()
+endforeach()
 
-    add_library(mkl::blas_interface UNKNOWN IMPORTED)
-    set_target_properties(mkl::blas_interface PROPERTIES IMPORTED_LOCATION ${MKL_INTERFACE_LIB})
+# Check if core libs were found
+#
+find_package_handle_standard_args(MKL REQUIRED_VARS MKL_INCLUDE_DIR
+                                                    MKL_CORE_LIB
+                                                    Threads_FOUND
+                                      HANDLE_COMPONENTS)
 
-    add_library(mkl::mkl INTERFACE IMPORTED)
-    set_target_properties(mkl::mkl PROPERTIES 
+if (MKL_FOUND AND NOT TARGET mkl::core)
```

Comment:
Is it necessary to have `mkl::core` and `mkl::blas` in separate targets? In case I would move this up in the file, so we can re-use it as a dependency for others, in place of ${MKL_CORE_LIB}.

Reply:
I am not sure if they need to be separate. I moved it up in the file in my latest commit.